controller/linodemachine_controller.go

```diff
@@ -317,30 +320,40 @@ func (r *LinodeMachineReconciler) reconcileCreate(
 		linodeInstance = &linodeInstances[0]
 	case 0:
 		// get the bootstrap data for the Linode instance and set it for create config
-		createConfig, err := r.newCreateConfig(ctx, machineScope, tags, logger)
+		createOpts, err := r.newCreateConfig(ctx, machineScope, tags, logger)
 		if err != nil {
 			logger.Error(err, "Failed to create Linode machine create config")
 
 			return nil, err
 		}
 
 		if machineScope.LinodeCluster.Spec.VPCRef != nil {
-			iface, err := r.getVPCInterfaceConfig(ctx, machineScope, createConfig.Interfaces, logger)
+			iface, err := r.getVPCInterfaceConfig(ctx, machineScope, createOpts.Interfaces, logger)
 			if err != nil {
 				logger.Error(err, "Failed to get VPC interface config")
 
 				return nil, err
 			}
 
-			createConfig.Interfaces = append(createConfig.Interfaces, *iface)
+			createOpts.Interfaces = append(createOpts.Interfaces, *iface)
 		}
 
-		linodeInstance, err = machineScope.LinodeClient.CreateInstance(ctx, *createConfig)
+		linodeInstance, err = machineScope.LinodeClient.CreateInstance(ctx, *createOpts)
 		if err != nil {
 			logger.Error(err, "Failed to create Linode machine instance")
 
 			return nil, err
 		}
+
+		// create etcd disk only if this is a control plane node
```

Comment:
nit: remove etcd from comment. Refer to it as "control plane disk configuring"

And the r.configureDisks could take a param for control plane, or probably better be configureDisksControlPlane so that we can future proof.

Likely, we are going to want to do different partitions for either types, when we talk about upgrades or immutable OS where we could potentially use different partitions to reboot in the OS, and have separate disk configurations that we can boot.